library/oneview_certificates_server.py

```diff
@@ -0,0 +1,145 @@
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.validateetag
+'''
+
+EXAMPLES = '''
+- name: Create a Server Certificate
+  oneview_certificates_server:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 1200
+    state: present
+    name: "172.18.13.11"
+    data:
+      certificateDetails:
+        - aliasName: 'vcenter'
+          base64Data: '--- Certificate ---'
+
+- name: Update the Server Certificate name to 'vcenter Renamed'
+  oneview_certificates_server:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 1200
+    state: present
+    name: "172.18.13.11"
+    data:
+      name: 'vcenter renamed'
+      certificateDetails:
+        - aliasName: 'vcenter'
+          base64Data: '--- Certificate ---'
+
+- name: Ensure that the Hypervisor Manager is absent
+  oneview_certificates_server:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 1200
+    state: absent
+    name: "172.18.13.11"
+    data:
+      alias_name: 'vcenter'
+'''
+
+RETURN = '''
+certificate_server:
+    description: Has the facts about the managed OneView Hypervisor Manager.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible.module_utils.oneview import OneViewModule
+
+
+class CertificatesServerModule(OneViewModule):
+    MSG_CREATED = 'Server Certificate created successfully.'
+    MSG_UPDATED = 'Server Certificate updated successfully.'
+    MSG_DELETED = 'Server Certificate deleted successfully.'
+    MSG_ALREADY_PRESENT = 'Server Certificate is already present.'
+    MSG_ALREADY_ABSENT = 'Server Certificate is already absent.'
+    RESOURCE_FACT_NAME = 'certificate_server'
+
+    def __init__(self):
+        additional_arg_spec = dict(data=dict(required=True, type='dict'),
+                                   name=dict(required=False, type='str'),
+                                   state=dict(
+                                       required=True,
+                                       choices=['present', 'absent']))
+
+        super(CertificatesServerModule, self).__init__(additional_arg_spec=additional_arg_spec, validate_etag_support=True)
+        self.__set_current_resource(self.oneview_client.certificates_server)
+
+    def execute_module(self):
+        if self.state == 'present':
+            return self.resource_present(self.RESOURCE_FACT_NAME)
+        elif self.state == 'absent':
+            return self.resource_absent()
+
+    def __set_current_resource(self, resource_client, aliasname=None):
+        self.resource_client = resource_client
+
+        if self.module.params.get('name'):
+            aliasname = self.module.params['name']
+
+        try:
+            self.current_resource = self.resource_client
+            self.current_resource.data = self.resource_client.get_by_aliasName(aliasname)
+        except Exception as e:
```

Comment:
get_by_aliaName method is not raising an exception if a certificate doesn't exist with the name passed, this method is just making a get_call and returning the result.

Reply:
(This is related to Python SDK, please open an issue for Python SDK to fix this)
method's name style is not consistent, should not mix camelcase and underscore in one name.
And this method should return the resource object.

Reply:
get_by_aliasName method raises an exception if the certificate doesn't exist with that name..

  File "/home/venkatesh/Documents/oneview-python/hpOneView/connection.py", line 338, in get
    raise HPOneViewException(body)
hpOneView.exceptions.HPOneViewException: ('Unable to retrieve the specified certificate. No matching certificate found for the specified alias.', {'errorCode': 'Cert.SSL_CERTIFICATE_MISSING_READ_ERROR', 'message': 'Unable to retrieve the specified certificate. No matching certificate found for the specified alias.', 'details': '', 'recommendedActions': ['Please specify valid alias name.'], 'errorSource': None, 'nestedErrors': [], 'data': {}})